addons/binding/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/discovery/internal/ModbusEndpointDiscoveryService.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.modbus.discovery.internal;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.openhab.binding.modbus.discovery.ModbusDiscoveryListener;
+import org.openhab.binding.modbus.discovery.ModbusDiscoveryParticipant;
+import org.openhab.binding.modbus.handler.ModbusEndpointThingHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A new instance of this class is created for each Modbus endpoint handler
+ * that supports discovery.
+ * This service gets called each time a discovery is requested, and it is
+ * responsible to execute the discovery on the connected thing handler.
+ * Actual discovery is done by the registered ModbusDiscoveryparticipants
+ *
+ * @author Nagy Attila Gabor - initial contribution
+ *
+ */
+public class ModbusEndpointDiscoveryService implements ModbusThingHandlerDiscoveryService {
```

Comment:
[Class should have @NonNullByDefault](https://www.openhab.org/docs/developer/development/conventions.html#null-annotations). Mark @Nullable fields and remove explicit @NonNull annotations

Reply:
done